Baikal/PostEffects/ML/denoiser_preprocess.h

```diff
@@ -0,0 +1,82 @@
+/**********************************************************************
+Copyright (c) 2019 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#pragma once
+
+#include "data_preprocess.h"
+
+#include "CLW.h"
+#include "Utils/clw_class.h"
+
+#ifdef BAIKAL_EMBED_KERNELS
+#include "embed_kernels.h"
+#endif
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+
+namespace Baikal
+{
+    namespace PostEffects
+    {
+        enum class Model
+        {
+            kColorDepthNormalGloss7,
+            kColorAlbedoNormal8,
+            kColorAlbedoDepthNormal9
+        };
+
+        class DenoiserPreprocess: public DataPreprocess
+        {
+        public:
+            DenoiserPreprocess(CLWContext context,
+                               CLProgramManager const* program_manager,
+                               std::uint32_t start_spp = 8);
+
+            Image MakeInput(PostEffect::InputSet const& inputs) override;
+
+            std::set<Renderer::OutputType> GetInputTypes() const override;
+
+            std::tuple<std::uint32_t, std::uint32_t> ChannelsNum() const override;
+        private:
+            void Init(std::uint32_t width, std::uint32_t height);
+
+            using MemoryLayout = std::vector<std::pair<Renderer::OutputType, std::size_t>>;
```

Comment:
Please add comment that size_t is channels num

Reply:
...and make it `int` :)